source/slang/compiler.cpp

```diff
@@ -301,25 +301,33 @@ namespace Slang
         char const* stagePrefix = nullptr;
         switch( profile.GetStage() )
         {
+            // Note: All of the raytracing-related stages require
```

Comment:
Perhaps there is an argument to have a 'library' stage. Unknown is typically an error case and it might make sense to leave it so.

Reply:
I think with the way the compiler is currently designed, it doesn't make sense to have a `Stage::Library` case, because we are increasingly shifting to a model where a `Stage` is a property specific to an entry point, and every entry point *needs* a stage right now (e.g., there is no way to write `[shader("library")]`, and have it mean anything).

The concept of a library, to me, amounts to wanting to compile a module with zero or more entry points (and maybe other exported non-entry-point symbols) into a single file, so the library-ness is not associated with any one entry point.

dxc kind of mirrors this, although in a way that is more obscure. There are basically two modes in which you can invoke dxc:

1. Specify a profile that includes a stage (e.g., `vs_5_0`) and a single entry point name (e.g., `vsMain`) and you'll get DXIL for one entry point (effectively, a library with a single entry point)

2. Specify a `lib_*` profile, and no entry point name (or at least the entry point name seems to get ignored), and you'll get DXIL for all the `[shader(...)]`-marked entry points in your file.

The gotcha is that option (1) is arbitrarily not allowed for the DXR stages, so that Slang has to fake it by using option (2) but only having a single entry point in the HLSL code we pass along.

Slang supports more or less the same two options, with the big differences that:

* We support option (1) for *all* stages, including the DXR ones

* We support multiple explicit entry points for option (1), so that asking for "library" output and specifying your stages manually are not mutually exclusive.

The Slang model is thus more general, and makes it clear that asking for a "library" output is orthogonal from the specification of stages.